File: Pythonleetcode/Leetcode530.py
class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def getMinimumDifference(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root or (not root.left and not root.right):
            return 0
        val_list=[]
        temp_node=[root]
        while len(temp_node)>0:
            a=temp_node.pop()
            val_list.append(a.val)
            if a.right:
                temp_node.append(a.right)
            if a.left:
                temp_node.append(a.left)
        val_list.sort()
        ans=10**9
        for i in range(len(val_list)-1):
            k=abs(val_list[i+1]-val_list[i])
            if k<ans:
                ans=k
        return ans

File: JavaScriptPractice/Leetcode530.js
let nums;
var helper=function(node)
{
    if(node==null){return;}
    nums.push(node.val);
    helper(node.left);
    helper(node.right);
    return;

}
var getMinimumDifference = function(root) 
{
    nums=[];
    let ans=Infinity;
    let temp=0;
    helper(root);
    nums.sort((a,b)=>a-b);
    for(let i=1;i<nums.length;i++)
    {
        temp = nums[i]-nums[i-1];
        if(temp<ans){ans=temp;}
    }
    return ans;
};